File: solutions/007/007.go
package main

import (
	"fmt"
)

func main() {
	n := 10001
	primes := []int{2}
	for x := 3; len(primes) < n+1; x += 2 {
		for _, p := range primes {
			if x%p == 0 { // Not prime
				break
			} else if p*p > x { // Is prime
				primes = append(primes, x)
				break
			}
		}
	}
	fmt.Printf("%d\n", primes[10000])
}

File: solutions/007/007.py
n = 10001
primes = [2]
current = 3
while len(primes) < n:
    for prime in primes:
        if current % prime == 0:
            break
        elif prime ** 2 > current:
            primes.append(current)
            break
    current += 2
print(primes[-1])
